src/category_theory/abelian/homology.lean

```diff
@@ -261,5 +264,54 @@ begin
 end
 
 end
-
 end homology
+
+namespace category_theory.functor
+
+variables {ι : Type*} {c : complex_shape ι} {B : Type*} [category B] [abelian B] (F : A ⥤ B)
+  [functor.additive F] [preserves_finite_limits F] [preserves_finite_colimits F]
+
+/-- When `F` is an exact additive functor, `F(Hᵢ(X)) ≅ Hᵢ(F(X))` for `X` a complex. -/
+noncomputable def map_homology_iso (C : homological_complex A c) (j : ι) :
+  F.obj (C.homology j) ≅ ((F.map_homological_complex _).obj C).homology j :=
+(preserves_cokernel.iso _ _).trans (cokernel.map_iso _ _ ((F.map_iso (image_subobject_iso _)).trans
+  ((preserves_image.iso _ _).symm.trans (image_subobject_iso _).symm))
+  ((F.map_iso (kernel_subobject_iso _)).trans ((preserves_kernel.iso _ _).trans
+  (kernel_subobject_iso _).symm))
+  begin
+    dsimp,
+    ext,
+    simp only [category.assoc, image_to_kernel_arrow],
+    erw [kernel_subobject_arrow', kernel_comparison_comp_ι, image_subobject_arrow'],
+    simp [←F.map_comp],
+  end)
+
+lemma homology_map_comp_map_homology_iso
```

Comment:
This lemma is exactly the "sorry" in the following definition:
```lean
variable (c)
@[simps]
def homology_functor_iso (j : ι) :
  homology_functor A c j ⋙ F ≅ F.map_homological_complex c ⋙ homology_functor B c j :=
nat_iso.of_components (λ X, F.map_homology_iso X j)
  (λ X Y f, sorry)
```
It would be better not to have a separate lemma `homology_map_comp_map_homology_iso`: put it in the definition of `homology_functor_iso`, so that if would become `(F.homology_functor_iso c j).hom.naturality`.

(Oops, I had not seen that you were defining this natural iso in the following lines. Anyway, I think it is better not to have a separate lemma.)